### DISCUSSION TITLE
How to get coverage results to work with `import.meta.env...`?

### DISCUSSION THREAD

**Discussion by jongeyer** [vitest-dev/vitest#1824] [Q&A] [answered]
I see that #1148 and #1149 had relevant discussion on using `import.meta.var` syntax in spec files. I have this working correctly.

But when trying to get coverage results with the `c8` package any `import.meta.env...` statement in the covered file is causing the coverage results to show as 100% covered -- obviously incorrect.

- I see correct coverage results in files that have no `import.meta.env...`.
- All the env vars have the `VITE_` prefix.

Anyone aware of this and a solution?

Is this a bug / question that should be raised in https://github.com/bcoe/c8??

**Comment by jongeyer** [comment, marked as answer]
Quick update on this:
- I tried using the new package `@vitest/coverage-c8` but ran into the same problem. ❌ 
- I then tried the alternative `@vitest/coverage-istanbul` and got it to work without any problems ✅ 
  - Docs need to be updated to make it clearer how to use this option.

**Reply by sheremet-va** [reply]
JFI: Vitest transforms `import.meta.env` to `process.env`, so you could rewrite it in runtime.
